crates/ruff_python_formatter/src/statement/stmt_if.rs

```diff
@@ -1,12 +1,127 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
-use ruff_formatter::{write, Buffer, FormatResult};
-use rustpython_parser::ast::StmtIf;
+use crate::comments::{leading_alternate_branch_comments, trailing_comments, SourceComment};
+use crate::expression::parentheses::Parenthesize;
+use crate::prelude::*;
+use crate::FormatNodeRule;
+use ruff_formatter::{write, FormatError};
+use rustpython_parser::ast::{Ranged, Stmt, StmtIf, Suite};
 
 #[derive(Default)]
 pub struct FormatStmtIf;
 
 impl FormatNodeRule<StmtIf> for FormatStmtIf {
     fn fmt_fields(&self, item: &StmtIf, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let comments = f.context().comments().clone();
+
+        let mut current = IfOrElIf::If(item);
+        let mut else_comments: &[SourceComment];
+        let mut last_node_of_previous_body = None;
+
+        loop {
+            let current_statement = current.statement();
+            let StmtIf {
+                test, body, orelse, ..
+            } = current_statement;
+
+            let first_statement = body.first().ok_or(FormatError::SyntaxError)?;
+            let trailing = comments.dangling_comments(current_statement.into());
+
+            let trailing_if_comments_end = trailing
+                .partition_point(|comment| comment.slice().start() < first_statement.start());
+
+            let (if_trailing_comments, trailing_alternate_comments) =
+                trailing.split_at(trailing_if_comments_end);
+
+            if current.is_elif() {
+                let elif_leading = comments.leading_comments(current_statement.into());
+                // Manually format the leading comments because the formatting bypasses `NodeRule::fmt`
+                write!(
+                    f,
+                    [
+                        leading_alternate_branch_comments(elif_leading, last_node_of_previous_body),
+                        source_position(current_statement.start())
+                    ]
+                )?;
+            }
+
+            write!(
+                f,
+                [
+                    text(current.keyword()),
+                    space(),
+                    test.format().with_options(Parenthesize::IfBreaks),
+                    text(":"),
+                    trailing_comments(if_trailing_comments),
+                    block_indent(&body.format())
+                ]
+            )?;
+
+            else_comments = trailing_alternate_comments;
+            last_node_of_previous_body = body.last();
+
+            if let Some(elif) = else_if(orelse) {
```

Comment:
Does this mean there's no elif node, just if?

Reply:
Yep. `elif` is represented as an `else` body that contains a single `if` statement. Rome (and Roslyn) model this very similarly where the `alternate` (else) is either an `if` statement (in which it is an `else if`) or a block (an else).